secure/net/src/lib.rs

```diff
@@ -239,24 +276,49 @@ mod test {
     }
 
     #[test]
-    fn test_shutdown() {
+    fn test_client_shutdown() {
         let server_port = utils::get_available_port();
         let server_addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), server_port);
         let mut server = NetworkServer::new(server_addr);
-        let mut client = NetworkClient::connect(server_addr).unwrap();
+        let mut client = NetworkClient::new(server_addr);
 
         let data = vec![0, 1, 2, 3];
         client.write(&data).unwrap();
         let result = server.read().unwrap();
         assert_eq!(data, result);
 
         client.shutdown().unwrap();
-        let mut client = NetworkClient::connect(server_addr).unwrap();
+        let mut client = NetworkClient::new(server_addr);
         assert!(server.read().is_err());
 
         let data = vec![4, 5, 6, 7];
         client.write(&data).unwrap();
         let result = server.read().unwrap();
         assert_eq!(data, result);
     }
+
+    #[test]
+    fn test_server_shutdown() {
```

Comment:
isn't this commit belong closer to the first one?

Reply:
I first verified I could perform restarts with the client in the earlier commits. Then I verified I could handle restarts with the server. Thus I separated them into separate commits.